<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Read trimming - McGill SI 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">McGill SI 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Day 1 <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../data-dna/">Training data set</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Day 2 <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Day 4 <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistance genes</a>
</li>

                    
                        
<li >
    <a href="../kraken/">Sample identification with Kraken</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Extra <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li class="active">
    <a href="./">Read trimming</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                        
<li >
    <a href="../artemis/">Artemis</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../kraken/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../workflows/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#trimming-reads">Trimming reads</a></li>
        
            <li><a href="#introduction">Introduction</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#start">Start</a></li>
        
            <li><a href="#trimmomatic-functions">Trimmomatic functions</a></li>
        
            <li><a href="#run-trimmomatic">Run Trimmomatic</a></li>
        
            <li><a href="#examine-output">Examine output</a></li>
        
            <li><a href="#what-next">What next?</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="trimming-reads">Trimming reads</h1>
<h2 id="introduction">Introduction</h2>
<p>After checking your input sequence reads for quality (e.g. using FastQC) it might be necessary to trim the reads. Here we will use the Trimmomatic tool. For more inforamtion, see the Trimmomatic <a href="http://www.usadellab.org/cms/index.php?page=trimmomatic">webpage</a> and the <a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf">manual.</a></p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>input sequence reads to Trimmomatic</li>
<li>trim using appropriate parameters, and</li>
<li>examine the output trimmed reads.</li>
</ol>
<h2 id="start">Start</h2>
<ul>
<li>open your Galaxy instance</li>
<li>find your quality-checked Illumina sequence reads</li>
<li>e.g. <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn></li>
<li>We want to trim the parts of the reads that are of low quality</li>
<li>based on the FastQC results, we might want to trim the reads like this:<ul>
<li>trim Illumina adapters</li>
<li>leading and trailing bases - trim if quality is below 15</li>
<li>sliding window - trim once average quality is below 20</li>
</ul>
</li>
</ul>
<h2 id="trimmomatic-functions">Trimmomatic functions</h2>
<ul>
<li>
<p>Adapter trimming</p>
<ul>
<li>This function trims adapters, barcodes and other contaminants from the reads.</li>
<li>You need to supply a FASTA file of possible adapter sequences, barcodes etc to trim. See Trimmomatic website for detailed instructions.</li>
<li>The default quality settings are sensible.</li>
<li>This should always be the first trimming step if it is used.</li>
</ul>
</li>
<li>
<p>Sliding window trimming</p>
<ul>
<li>This function uses a sliding window to measure average quality and trims accordingly.</li>
<li>The default quality parameters are sensible for this step.</li>
</ul>
</li>
<li>
<p>Trailing bases quality trimming</p>
<ul>
<li>This function trims bases from the end of a read if they drop below a quality threshold. e.g. If base 69 of 75 drops below the threshold, the read is cut to 68 bases.</li>
<li>Use FastQC report to decide whether this step is warranted and what quality value to use. A quality threshold value of 10-15 is a good starting point.</li>
</ul>
</li>
<li>
<p>Leading bases quality trimming</p>
<ul>
<li>This function works in a similar fashion to trailing bases trimming except it performs it at the start of the reads.</li>
<li>Use FastQC report to determine if this step is warranted. If the quality of bases is poor at the beginning of reads it might be necessary.</li>
</ul>
</li>
<li>
<p>Minimum read length</p>
<ul>
<li>Once all trimming steps are complete, this function makes sure that the reads are still longer than this value. If not, the read is removed from the file and its pair is put into the orphan file.</li>
<li>The most appropriate value for this parameter will depend on the FastQC report, specifically the length of the high quality section of the Per Base Sequence Quality graph.</li>
</ul>
</li>
<li>
<p>Each read library should be trimmed separately with parameters dependent on their own FastQC reports.</p>
</li>
</ul>
<h2 id="run-trimmomatic">Run Trimmomatic</h2>
<!---
FIXME: change these settings if required (examine FastQC reports)
--->

<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: QC and manipulation &rarr; Trimmomatic</ss>.</li>
<li><ss>Input FASTQ file (R1/first of pair)</ss>: <fn>mutant_R1.fastq</fn></li>
<li><ss>Input FASTQ file (R2/second of pair)</ss>: <fn>mutant_R2.fastq</fn></li>
<li><ss>Perform initial ILLUMINACLIP step </ss>: <em>Yes</em>
&lt;!&mdash;</li>
<li><ss>Adapter sequences to use</ss>: FIXME
&mdash;&gt;</li>
<li><ss>How accurate the match between the two &lsquo;adapter ligated&rsquo; reads must be for PE palindrome read alignment</ss>: 40</li>
<li><ss>How accurate the match between any adapter etc. sequence must be against a read</ss>: 15</li>
<li>leave the first <ss>Trimmomatic Operation</ss> as is</li>
<li>click on <ss>+ Insert Trimmomatic Operation</ss></li>
<li><ss>Select Trimmomatic operation to perform</ss>: <em>Cut bases off the start of a read, if below a threshold quality (LEADING)</em></li>
<li><ss>Minimum quality required to keep a base</ss>: 15</li>
<li>click on <ss>+ Insert Trimmomatic Operation</ss></li>
<li><ss>Select Trimmomatic operation to perform</ss>: <em>Cut bases off the end of a read, if below a threshold quality (TRAILING)</em></li>
<li><ss>Minimum quality required to keep a base</ss>: 15</li>
<li>click <ss>Execute</ss></li>
</ul>
<!---
FIXME: screenshot of these trimmomatic options selected
--->

<h2 id="examine-output">Examine output</h2>
<p>Trimmomatic should produce 2 pairs files (1 left and 1 right hand end) and 1 or 2 single “orphaned reads” files.</p>
<p>The output files are the ones you should use for assembly.</p>
<p>There are four output files, still in FASTQ format:</p>
<ul>
<li>R1 reads that have a pair in the R2 file</li>
<li>R2 reads that have a pair in the R1 file</li>
<li>R1 reads with no pair (R2 match was low quality: deleted)</li>
<li>R2 reads with no pair (R1 match was low quality: deleted)</li>
</ul>
<p>Examine each file with the eye icon. Look for:</p>
<ul>
<li>Number of reads orphaned by the trimming / cleanup process.</li>
<li>Number of pairs lost totally.</li>
</ul>
<h2 id="what-next">What next?</h2>
<p>Next: use the output FASTQ files for Assembly, e.g. with <a href="../spades/">Spades</a></p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
                <center>&copy; The authors; 2016-</center>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>
        <script src="../../js/details.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>